tests/test_free_energy.py

```diff
@@ -373,3 +373,31 @@ def low_dim_f(perturb: float) -> float:
             for perturb in perturbations:
                 abs_err = check_grad(low_dim_f, grad(low_dim_f), perturb, epsilon=1e-4)
                 assert abs_err < 1e-3
+
+
+def test_consruct_differentiable_interface_cpp():
+    """Assert that the computation of U and its derivatives using the
+    C++ code path produces equivalent results to the doing the
+    summation in Python"""
+
+    ff_params = hif2a_ligand_pair.ff.get_ordered_params()
+    unbound_potentials, sys_params, _, coords = hif2a_ligand_pair.prepare_vacuum_edge(ff_params)
+    box = np.eye(3) * 100
+    lam = 0.5
+
+    for precision in [np.float32, np.float64]:
+        U = construct_differentiable_interface(unbound_potentials, precision)
+        U_cpp = construct_differentiable_interface_cpp(unbound_potentials, sys_params, precision)
+        args = (coords, sys_params, box, lam)
+        assert np.allclose(U(*args), U_cpp(*args), rtol=1e-6)
```

Comment:
Would we prefer `np.testing.assert_arrays_equal` here if possible?

Reply:
For this test, I'd expect the results here to be bitwise identical.

Reply:
Actually - this is a little more complicated than I thought, since I think the old code actually does:

`
sum_floats(fixed_to_float(U_fixed(x_i)))
`

Where as the new code does

`
fixed_to_float(sum_ints(U_fixed(x_i)))
`

So I'm not fully sure that this commutes.

Reply:
Hmm, it does seem to pass with the equality assertion (9a32491). In light of your comment I'm not sure I understand how this works; will look into it more.